packages/core/useCloned/index.ts

```diff
@@ -0,0 +1,41 @@
+import type { MaybeRef } from '@vueuse/shared'
+import type { WatchOptions } from 'vue-demi'
+import { isRef, ref, unref, watch } from 'vue-demi'
+
+export interface UseClonedOptions<T extends Record<any, any> = Record<any, any>> {
+  /**
+   * sync source only by function
+   */
+  manual?: boolean
+  /**
+   * Custom clone function should return new value for cloned data
+   */
+  cloneFunction?: (source: T, cloned: T) => T
+  /**
+   * Options for watcher
+   *
+   * @default { immediate: true, deep: true }
+   */
+  watchOptions?: WatchOptions
+}
+
+export function useCloned<T extends Record<any, any> = Record<any, any>>(source: MaybeRef<T>, options: UseClonedOptions = {}) {
+  const cloned = ref<T>({} as T)
+
+  const { manual, cloneFunction, watchOptions = { immediate: true, deep: true } } = options
+
+  if (!manual && isRef(source))
+    watch(source, sync, watchOptions)
+  else
+    sync()
+
+  function defaultCloning() {
```

Comment:
actually there's a native deep clone api called: [structuredClone](https://caniuse.com/?search=structuredClone), and supported quite well on different browsers. Use it if exists on `window`.

Reply:
> actually there's a native deep clone api called: [structuredClone](https://caniuse.com/?search=structuredClone), and supported quite well on different browsers. Use it if exists on `window`.

Hm, I like it, I'll add this one. Thank you!

Reply:
`structuredClone` is not available in all browsers. We should let users pass it explicitly instead of using the current fallback, which will lead to different behavior on different browsers implicitly.

Reply:
> `structuredClone` is not available in all browsers. We should let users pass it explicitly instead of using the current fallback, which will lead to different behavior on different browsers implicitly.

removed `structuredClone`